pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -232,14 +277,38 @@ abstract contract VaultActions is IBaseRelayerLibrary {
         }
     }
 
+    /**
+     * @dev Compute the final userData for an exit, depending on the PoolKind, performing replacements for chained
+     * references as necessary.
+     *
+     * While all Stable Pool versions fortuitously support the same join kinds (V2 and higher support one extra),
+     * they do NOT all support the same exit kinds. Also, though the encoding of the data associated with the exit
+     * is uniform across pool kinds for the same exit method, the ExitKind ID itself may have a different value.
+     *
+     * For instance, BPT_IN_FOR_EXACT_TOKENS_OUT is 2 in legacy Stable Pools, but 1 in Composable Stable Pools.
+     * (See the reference comment and libraries below.)
+     *
+     * To accommodate this, we delegate first to custom ChainedReferenceReplacement functions for each pool type.
+     * These functions determine the general exit method (e.g., single or multi-token), and then - since the encoding
+     * of the arguments is the same for each method across pool kinds - forward the specific ExitKind ID appropriate
+     * to that pool kind to the shared functions that perform the actual recoding.
```

Comment:
I'd move this inside the body of the function

Reply:
Also I'm not sure we need to explain all of this here? I think we could not have this comment and simply explain in `_doStableExactBptInForOneTokenOutReplacements` and `_doStableExactBptInForTokensOutReplacements` that those kinds have different enum values depending on the pool version, which is why they receive it as an argument and use to to re-encode, instead of hardcoding a constant.

Reply:
Ok; I've distributed the content to lower levels, instead of concentrating it all here. This just has the top sentence, saying what the overall process is doing.

Then I have a section for "Stable Pool version-dependent recoding dispatch functions," which extract the exitKind and have version-dependent logic. That explains the middle part (why there have to be separate functions for each pool version).

Then the "Shared Stable Exit recoding functions" second has an abbreviated explanation of why ExitKind has to be passed in. See how you like it now.